aten/src/ATen/native/Loss.cpp

```diff
@@ -148,4 +154,48 @@ Tensor poisson_nll_loss(const Tensor& input, const Tensor& target, const bool lo
 
     return apply_loss_reduction(loss, reduction);
 }
+
+Tensor mse_loss(const Tensor& input, const Tensor& target, int64_t reduction) {
+  Tensor loss;
+  auto iter = TensorIterator::binary_op(loss, input, target);
+  mse_stub(iter.device_type(), iter);
+  return apply_loss_reduction(iter.output(), reduction);
+}
+
+Tensor& mse_loss_out(Tensor&result, const Tensor& input, const Tensor& target, int64_t reduction) {
+  if (reduction != Reduction::None) {
+    Tensor loss;
+    auto iter = TensorIterator::binary_op(loss, input, target);
+    mse_stub(iter.device_type(), iter);
+    if (reduction == Reduction::Mean) {
+      at::mean_out(result, iter.output(), 0);
+    } else {
+      at::sum_out(result, iter.output(), 0);
+    }
+  } else {
+    auto iter = TensorIterator::binary_op(result, input, target);
+    mse_stub(iter.device_type(), iter);;
+  }
+  return result;
+}
+
+Tensor mse_loss_backward(const Tensor& grad_output, const Tensor& input, const Tensor& target, int64_t reduction) {
+  Tensor grad_input = at::zeros_like(input);
+  return at::mse_loss_backward_out(grad_input, grad_output, input, target, reduction);
+}
+
+Tensor& mse_loss_backward_out(Tensor& grad_input, const Tensor& grad_output,
+    const Tensor& input, const Tensor& target, int64_t reduction) {
+  auto norm = reduction == Reduction::Mean ? 2. / input.numel() : 2.;
```

Comment:
@zou3519 , there have a problem for mse_backward if delete **broadcast_tensors**,  for input.size=(2, 1), target.size =(2,10), the norm number should be 20, but get 2 in there.  So I didn't remove this **broadcast_tensors** method now.

Reply:
Oh, I see... It's because we have a custom backward implementation for it. That's fine for now then, we can work on improving `mse_loss` separately after the port, thank you for pointing that out.